src/middlewared/middlewared/plugins/service_/services/iscsitarget.py

```diff
@@ -14,6 +14,12 @@ class ISCSITargetService(SimpleService):
 
     systemd_unit = "scst"
 
+    async def after_start(self):
+        await self.middleware.call("iscsi.host.injection.start")
```

Comment:
I dont recall how the services are started in SCALE. Are these methods triggered when service is started by systemd?

Reply:
No, they are not. We rely on the fact that user does not use scripts or systemctl to start/stop services and always does so through the middleware.

Reply:
Ok, but what starts scst at boot?

Reply:
Systemd starts it. And we have this https://github.com/freenas/freenas/pull/5979/files#diff-15237d4d3fd65ead3be91fce6291c116db8681d11e69f8afdc52eb79a06295b2R106

Reply:
I am a bit confused, I am rusty, sorry :)

How does that diff you pasted helps? middlewared starts before scst so doing it on setup should be a no-op unless we restart middleware

Reply:
The condition there is `service["enable"] or service["state"] == "RUNNING":`

`service["enable"]` — this means that systemd will start scst shortly
`service["state"] == "RUNNING":` — this is mostly for us, developers, when we restart middleware after the system has already booted

Reply:
Ah makes sense, thanks for explaining